tests/MongoDB.Driver.Core.Tests/Specifications/read-write-concern/ConnectionStringTestRunner.cs

```diff
@@ -76,13 +72,28 @@ private void AssertValid(ConnectionString connectionString, BsonDocument definit
             BsonValue writeConcernValue;
             if (definition.TryGetValue("writeConcern", out writeConcernValue))
             {
-                var writeConcern = WriteConcern.FromBsonDocument(MassageWriteConcernDocument((BsonDocument)writeConcernValue));
+                try
+                {
+                    var writeConcern =
+                        WriteConcern.FromBsonDocument(MassageWriteConcernDocument((BsonDocument) writeConcernValue));
+                    connectionString.W.Should().Be(writeConcern.W);
+                    connectionString.WTimeout.Should().Be(writeConcern.WTimeout);
+                    connectionString.Journal.Should().Be(writeConcern.Journal);
+                    connectionString.FSync.Should().Be(writeConcern.FSync);
+                }
+                //if invalid parameter combination
+                catch (Exception ex)
```

Comment:
This method is called `AssertValid`. We should be expecting the connection string to be valid and not catching exceptions.

I think some refactoring is needed.

Reply:
With @ksadoff 's addition of the check to the `ConnectionString.Parse()` cleanly obviating the need to change the test runner while still allowing the tests to pass (and thus her reverting the changes), I think we should be able to resolve this thread? @rstam